src/popup/router/pages/Retip.vue

```diff
@@ -34,69 +37,97 @@
 
 <script>
 import { mapGetters, mapState } from 'vuex';
-import BigNumber from 'bignumber.js';
-import tipping from 'aepp-raendom/src/utils/tippingContractUtil';
+import tipping from 'tipping-contract/util/tippingContractUtil';
 import { MAGNITUDE, calculateFee, TX_TYPES } from '../../utils/constants';
+import { convertToken } from '../../utils/helper';
 import deeplinkApi from '../../../mixins/deeplinkApi';
 import AmountSend from '../components/AmountSend';
 import UrlStatus from '../components/UrlStatus';
 import Button from '../components/Button';
+import BalanceInfo from '../components/BalanceInfo';
 
 export default {
   mixins: [deeplinkApi],
-  components: { AmountSend, UrlStatus, Button },
+  components: { AmountSend, UrlStatus, Button, BalanceInfo },
   data: () => ({
     tip: {},
     amount: null,
     amountError: false,
     loading: false,
     minCallFee: null,
+    tippingContract: null,
   }),
   computed: {
     ...mapGetters(['account', 'tippingSupported', 'minTipAmount', 'activeNetwork']),
-    ...mapState(['balance', 'tippingV1', 'sdk']),
+    ...mapState('fungibleTokens', ['selectedToken']),
+    ...mapState(['balance', 'tippingV1', 'tippingV2', 'sdk']),
     urlStatus() {
       return this.$store.getters['tipUrl/status'](this.tip.url);
     },
+    shouldUseV2Contract() {
+      return this.$route.query.id.includes('_v2') || this.$route.query.id.includes('_v3');
+    },
   },
   watch: {
     amount() {
-      this.amountError = !+this.amount || this.amount < this.minTipAmount;
+      this.amountError = !+this.amount || (!this.selectedToken && this.amount < this.minTipAmount);
     },
   },
   async created() {
     this.loading = true;
     await this.$watchUntilTruly(() => this.sdk);
     this.minCallFee = calculateFee(TX_TYPES.contractCall, {
       ...this.sdk.Ae.defaults,
-      contractId: this.activeNetwork.tipContractV1,
+      contractId: this.shouldUseV2Contract
+        ? this.activeNetwork.tipContractV2
+        : this.activeNetwork.tipContractV1,
```

Comment:
Let's get the address from contract to don't depend on `activeNetwork`:
```suggestion
      contractId: this.tippingContract.deployInfo.address,
```